src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpectatorView.cs

```diff
@@ -206,5 +238,70 @@ private bool TryCreateRecordingService(out IRecordingService recordingService)
             return false;
 #endif
         }
+
+        private void DebugLog(string message)
+        {
+            if (debugLogging)
+            {
+                UnityEngine.Debug.Log($"SpatialLocalizationInitializationSettings: {message}");
+            }
+        }
+
+        private async void OnParticipantConnected(SpatialCoordinateSystemParticipant participant)
+        {
+            DebugLog($"Waiting for the set of supported localizers from connected participant {participant.SocketEndpoint.Address}");
+
+            // When a remote participant connects, get the set of ISpatialLocalizers that peer
+            // supports. This is asynchronous, as it comes across the network.
+            ISet<Guid> peerSupportedLocalizers = await participant.GetPeerSupportedLocalizersAsync();
+
+            // If there are any supported localizers, find the first configured localizer in the
+            // list that supports that type. If and when one is found, use it to perform localization.
+            if (peerSupportedLocalizers != null)
+            {
+                DebugLog($"Received a set of {peerSupportedLocalizers.Count} supported localizers");
+                
+                if (SpatialLocalizationInitializationSettings.IsInitialized &&
+                    TryRunLocalization(SpatialLocalizationInitializationSettings.Instance.PrioritizedInitializers, peerSupportedLocalizers, participant))
+                {
+                    // Succeeded at using a custom localizer specified by the app.
+                    return;
+                }
+
+                if (TryRunLocalization(defaultSpatialLocalizationInitializers, peerSupportedLocalizers, participant))
+                {
+                    // Succeeded at using one of the default localizers from the prefab.
+                    return;
+                }
+
+                if (defaultSpatialLocalizationInitializers != null)
+
+                DebugLog($"None of the configured LocalizationInitializers were supported by the connected participant, localization will not be started");
+            }

```

Comment:
nit: we should log this failure regardless of whether or not the defaultSpatialLocalizationInitializers is null. Also, these failures may be worth escalating to Debug.Warning

Reply:
Whoa, that was totally not an intentional line of code to leave here, will get rid of the if.

---
In reply to: [298353301](https://github.com/microsoft/MixedReality-SpectatorView/pull/48#discussion_r298353301) [](ancestors = 298353301)